src/state/workload.rs

```diff
@@ -614,6 +614,13 @@ impl WorkloadStore {
             self.by_hostname.insert(w.hostname.clone(), w.clone());
         }
         self.by_uid.insert(w.uid.clone(), w.clone());
+        // Only track local nodes to avoid overhead
+        if track_identity {
+            self.by_identity
+                .entry(w.identity())
+                .or_default()
+                .insert(w.uid.clone());
+        }
```

Comment:
I think this comment describes the intended use for track_identity rather than what this code does.  Should we consider adding this comment to the insert method itself and/or changing `track_identity` to be `is_node_local` perhaps?

Reply:
`should_track_certificates_for_removal()` probably belongs in `workload` and should be checked as part of `insert`, and not in `cert_mgr` - it's being called before `workload.insert` pretty much everywhere.

Reply:
The reason I did it this way is workload doesn't have access to the proxy mode, local node, etc at all, so it cannot make the call itself. We could plumb it down, but then its inconsistent with the `should_prefetch_certificate` and requires even more plumbing around

Reply:
> The reason I did it this way is workload doesn't have access to the proxy mode, local node, etc at all,

Figured.

> We could plumb it down, but then its inconsistent with the should_prefetch_certificate and requires even more plumbing around

`should_prefetch` and `should_track` are almost the same check and it feels like _both_ should probably go in `workload` but yeah that's getting OOS for this.